File: chap11/RepertoireFichier.cpp
// Ces programmes sont sous licence CeCILL-B V1.
#include <iostream>
#include <fstream>
#include <string>

using namespace std;

const int TAILLE_REPERTOIRE = 10;

int main() 
{
   string nom[TAILLE_REPERTOIRE];
   string tel[TAILLE_REPERTOIRE];
   int i;
   string s;

   ifstream f;
   f.open ("repertoire.txt", ifstream::in);
   for (i = 0; i <= TAILLE_REPERTOIRE - 1; i = i + 1) 
   {
      f >> nom[i];
      f >> tel[i];
   }
   f.close();
   cin >> s;
   i = 0;
   while (i < TAILLE_REPERTOIRE && s.compare(nom[i]) != 0)
   {
      i = i + 1;
   }
   if (i < TAILLE_REPERTOIRE)
   {
      cout << tel[i] << endl;
   }
   else 
   {
      cout << "Inconnu" << endl;
   }
}

File: chap11/RepertoireFichier.py
# Ces programmes sont sous licence CeCILL-B V1.
tailleRepertoire = 10
nom = [None]*tailleRepertoire
tel = [None]*tailleRepertoire

# Le codage peut être aussi "iso-8859-1" ou (beurk windows !) "cp1252"
f = open("repertoire.txt","r",encoding="utf-8")
# strip() retire les blancs devant et derrière, ici surtout 
# les caractères de fin de ligne que readline() laisse
for i in range(tailleRepertoire):
   nom[i] = f.readline().strip()
   tel[i] = f.readline().strip()
f.close()

s = input()
i = 0
while (i < 10) and not (s == nom[i]):
   i = i + 1
if i < 10:
   print(tel[i])
else:
   print("Inconnu")
